tests/ui/entry-point/return-ty-has-bound-vars.rs

```diff
@@ -0,0 +1,5 @@
+// issue-119209
+
+type Foo<'a> = impl PartialEq; //~ERROR `impl Trait` in type aliases is unstable
+
+fn main<'a>(_: &'a i32) -> Foo<'a> {} //~ERROR `main` function return type is not allowed to have generic parameters
```

Comment:
```suggestion
// issue-119209
fn main<'a>(_: &'a i32) -> &'a () { &() } //~ERROR `main` function return type is not allowed to have generic parameters
```

minimized it a bit more, no TAIT is needed